lib/modules/backupAndRestore.js

```diff
@@ -0,0 +1,78 @@
+addModule('backupAndRestore', function(module, moduleID) {
+	$.extend(module, {
+		moduleName: 'Backup & Restore',
+		category: 'About RES',
+		description: 'Backup and restore your Reddit Enhancement Suite settings.',
+		options: {
+			backup: {
+				type: 'button',
+				text: 'backup',
+				callback: backup,
+				description: 'Download a backup of your current RES settings.'
+			},
+			restore: {
+				type: 'button',
+				text: 'restore',
+				callback: restore,
+				description: 'Restore a backup of your RES settings. Warning: This will overwrite your current settings.'
+			}
+		}
+	});
+
+	function backup() {
+		// Generate copy of RES settings for download
+		var settings = JSON.stringify(RESStorage);
+		var link = document.createElement('a');
+		if (BrowserDetect.isChrome()) {
+			// Encoding can make the URI long enough to crash Chrome; it doesn't appear to care if it's unencoded
+			link.href = 'data:text/plain;charset=utf-8,' + settings;
```

Comment:
You can get around this by [storing the json in a `Blob`](http://jsfiddle.net/koldev/cw7w5/) and setting `.href` to the `Blob`'s URI instead.